src/binutil/server.rs

```diff
@@ -275,6 +309,29 @@ fn run_raft_server(pd_client: RpcClient, cfg: &TiKvConfig, security_mgr: Arc<Sec
         );
     }
 
+    // Start waiter manager and deadlock detector
+    if cfg.pessimistic_txn.enabled {
+        let waiter_mgr_runner = WaiterManager::new(DetectorScheduler::new(
+            detector_worker.as_ref().unwrap().scheduler(),
```

Comment:
There are several places need detector_worker's scheduler, you can get the scheduler first, and clone it where ever it is needed.